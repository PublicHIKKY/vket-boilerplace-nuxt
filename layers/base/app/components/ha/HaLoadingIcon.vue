<template>
  <div class="ha-loading-icon" />
</template>

<style lang="scss" scoped>
@use '#base/app/assets/styles/variables' as v;

.ha-loading-icon {
  animation: ha-loading-icon 0.7s infinite linear;
  border: 2px solid v.$white;
  border-bottom-color: transparent;
  border-radius: 9999px;
  height: 36px;
  inset: 0;
  margin: auto;
  position: absolute;
  width: 36px;
}

@keyframes ha-loading-icon {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
</style>
