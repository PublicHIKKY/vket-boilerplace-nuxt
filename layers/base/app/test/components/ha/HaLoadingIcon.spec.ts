import { mount } from '@vue/test-utils'
import { expect, test } from 'vitest'
import HaLoadingIcon from '#base/app/components/ha/HaLoadingIcon.vue'

test('ref component', () => {
  expect(HaLoadingIcon).toBeTruthy()
})

test('mount component', () => {
  const wrapper = mount(HaLoadingIcon)
  expect(wrapper.getCurrentComponent()).toBeTruthy()
  expect(wrapper.html()).toMatchSnapshot()
})
