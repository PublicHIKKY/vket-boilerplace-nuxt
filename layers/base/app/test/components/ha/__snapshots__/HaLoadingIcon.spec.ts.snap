// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component 1`] = `"<div data-v-c1d69a8c="" class="ha-loading-icon"></div>"`;
